File: Leetcode/9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        x=str(x)
        if len(x)<2:
            return True
        else:
            if len(x)%2==0:
                first=x[:len(x)//2]
                second=x[len(x)//2:]
                second= second[::-1]
                if first==second:
                    return True
                else:
                    return False
            else:
                first=x[:len(x)//2]
                second=x[(len(x)//2)+1:]
                second= second[::-1]
                if first==second:
                    return True
                else:
                    return False

File: Leetcode/9.cpp
#include<iostream>
#include<string>
#include <sstream>
using namespace std;
class Solution {
public:
    bool isPalindrome(int x) {
        stringstream st;
        st<<x;
        string val = st.str();
        if (val.length()<2){
            return true;
        }
        else{
            if (val.length()%2==0){
                string first= val.substr(0,val.length()/2);
                string second= val.substr(val.length()/2);
                for (int i = 0; i < second.length() / 2; i++)
                    swap(second[i], second[second.length() - i - 1]);
                if (first==second){
                    return true;
                }
                else{
                    return false;
                }

            }
            else{
                string first= val.substr(0,val.length()/2);
                string second= val.substr((val.length()/2)+1);
                for (int i = 0; i < second.length() / 2; i++)
                    swap(second[i], second[second.length() - i - 1]);
                if (first==second){
                    return true;
                }
                else{
                    return false;
                }
            }
        }
    }
};